prov/verbs/src/windows/verbs_nd_init.c

```diff
@@ -0,0 +1,751 @@
+/*
+ * Copyright (c) 2022 Canon Medical Research USA, Inc.  All rights reserved.
+ * Copyright (c) 2015-2016 Intel Corporation, Inc.  All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <assert.h>
+#include "ndspi.h"
+#include "ws2spi.h"
+#include "verbs_nd.h"
+
+int ofi_nd_is_valid_addr(const SOCKADDR *addr);
+int ofi_nd_addr_cmp(const void *vaddr1, const void *vaddr2);
+int ofi_nd_is_same_file(const wchar_t *path1, const wchar_t *path2);
+int ofi_nd_file_exists(const wchar_t *path);
+int ofi_nd_is_directory(const wchar_t *path);
+
+static size_t unique(void *base, size_t num, size_t width,
+		     int (*cmp)(const void *, const void *))
```

Comment:
The function parameters and name don't lend themselves to figuring out what this call does.

Reply:
This came straight from the netdir provider. Would a comment above the function like the following be sufficient?
```
/*
 * Assuming a sorted array,' base[num]', with elements of size 'width', this
 * function will compare adjacent elements using 'cmp()' and "compact" the
 * array to just its 'n' unique elements.
 */
```

Reply:
I would restructure to something like this then:
```
/* Array must be sorted */
static void ofi_nd_remove_dups(void *array, size_t *entries or count, size_t elem_size, ...)
```
This makes it a little clearer as to the intent.  There's still some confusion here.  If we're only removing duplicates and the array is sorted, why do we need the cmp function passed in?  memcmp would work